tests/shouldwork/Cores/Xilinx/XpmCdcGrayTypes.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE BangPatterns #-}
+
+module XpmCdcGrayTypes where
+
+import Clash.Cores.Xilinx.Xpm.Cdc.Gray
+import Clash.Explicit.Prelude
+import Clash.Explicit.Testbench
+import Data.Proxy
+import Language.Haskell.TH.Lib
+
+import qualified Prelude as P
+
+createDomain vXilinxSystem{vName="D3",  vPeriod=hzToPeriod 30e6}
+createDomain vXilinxSystem{vName="D5",  vPeriod=hzToPeriod 50e6}
+createDomain vXilinxSystem{vName="D11", vPeriod=hzToPeriod 110e6}
+
+noRst :: KnownDomain dom => Reset dom
+noRst = unsafeFromHighPolarity (pure False)
+
+dropN :: SNat 15
+dropN = SNat
+
+tb ::
+  forall a b width stages n .
+  ( KnownNat n, 1 <= n
+  , KnownNat stages, 2 <= stages, stages <= 10
+  , KnownNat width, 2 <= width, width <= 32
+  , KnownDomain a
+  , KnownDomain b
+  ) =>
+  Proxy a -> Proxy b -> SNat stages ->
+  -- | Expected data
+  Vec n (BitVector width) ->
+  Signal b Bool
+tb Proxy Proxy SNat expected = done
+ where
+  counter = delay clkA enableGen 0 (counter + 1)
+
+  actual = xpmCdcGrayWith @stages @width (XpmCdcGrayConfig SNat) clkA clkB counter
+
+  done =
+    outputVerifierWith
+      (\clk rst -> assertBitVector clk rst "outputVerifier Port A")
+      clkB clkB noRstB
+      (expected)
+      (pack <$> actual)
+
+  noRstA = noRst @a
```

Comment:
This isn't used in this function anywhere, is this some left over dead code?

Reply:
You're right, same story for `dropN`.